### ISSUE TITLE
Gateway Application startup fails

### ISSUE THREAD

**Issue by derehls** [jhipster/generator-jhipster#3265]
##### **Overview of the issue**

I checked out the jhipster-registry and started it via mvn spring-boot:run. 
Then i created a new microservice with one Entity "timeEntry". After that i created a new gateway application, linked to the "timeEntry" Entity in the microservice and try to start it via mvn spring-boot:run. The log says:

```
2016-03-25 20:32:34.015 ERROR 11777 --- [  restartedMain] o.s.boot.SpringApplication               : Application startup failed

java.lang.NoSuchMethodError: com.google.common.collect.Multimaps.asMap(Lcom/google/common/collect/ListMultimap;)Ljava/util/Map;
	at springfox.documentation.spring.web.scanners.ApiListingReferenceScanner.scan(ApiListingReferenceScanner.java:62)
	at springfox.documentation.spring.web.scanners.ApiDocumentationScanner.scan(ApiDocumentationScanner.java:65)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.scanDocumentation(DocumentationPluginsBootstrapper.java:105)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.onApplicationEvent(DocumentationPluginsBootstrapper.java:91)
	at springfox.documentation.spring.web.plugins.DocumentationPluginsBootstrapper.onApplicationEvent(DocumentationPluginsBootstrapper.java:53)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:163)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:136)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:381)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:335)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:855)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.finishRefresh(EmbeddedWebApplicationContext.java:140)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:541)
	at org.springframework.boot.context.embedded.EmbeddedWebApplicationContext.refresh(EmbeddedWebApplicationContext.java:118)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:766)
	at org.springframework.boot.SpringApplication.createAndRefreshContext(SpringApplication.java:361)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:307)
	at com.company.gateway.CompanyGatewayApp.main(CompanyGatewayApp.java:73)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.boot.devtools.restart.RestartLauncher.run(RestartLauncher.java:49)

```
##### **Motivation for or Use Case** 

Gateway App does not start

##### **JHipster Version(s)** 

3.0

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder** 

```
{
  "generator-jhipster": {
    "jhipsterVersion": "3.0.0",
    "baseName": "companyGateway",
    "packageName": "com.company.gateway",
    "packageFolder": "com/company/gateway",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "hibernateCache": "no",
    "clusteredHttpSession": "no",
    "websocket": "no",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": "no",
    "buildTool": "maven",
    "jwtSecretKey": "verySecret",
    "useSass": false,
    "applicationType": "gateway",
    "testFrameworks": [
      "gatling"
    ],
    "enableTranslation": false
  }
}
```

##### **Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory** 

```
{
    "relationships": [],
    "fields": [
        {
            "fieldName": "description",
            "fieldType": "String"
        },
        {
            "fieldName": "date",
            "fieldType": "LocalDate"
        },
        {
            "fieldName": "hours",
            "fieldType": "Integer"
        }
    ],
    "changelogDate": "20160324200647",
    "dto": "no",
    "service": "serviceClass",
    "entityTableName": "time_entry",
    "pagination": "no",
    "microserviceName": "timeEntry",
    "searchEngine": "no"
}
```

##### **Browsers and Operating System** 

Mint Linux 17.2
Oracle JDK 8u51
Maven 3.0.5

##### **Reproduce the error** 

Can be reproduced by creating a new gateway as described above.

##### **Possible solution**

Is there a dependency missing?

**Comment by derehls** [comment]
Fixed it by adding
```
<dependency>
            <groupId>com.google.guava</groupId>
            <artifactId>guava</artifactId>
            <version>18.0</version>
</dependency>
```
to pom

**Comment by deepu105** [comment]
I thought we removed the guava dependency

**Comment by deepu105** [comment]
We need to chevk this

**Comment by gmarziou** [comment]
I failed to reproduce but I found another unrelated issue: after having generated the entity on gateway, the bower dependencies were not injected into the index.html because the entity sub generator did call only `gulp inject` and not `gulp wiredep`

**Comment by gmarziou** [comment]
Regarding this issue, I ran `mvn dependency:tree` and found guava as a dependency of springfox-swagger2 with same version that you added.

~~~
[INFO] +- io.springfox:springfox-swagger2:jar:2.4.0:compile
[INFO] |  +- io.swagger:swagger-annotations:jar:1.5.6:compile
[INFO] |  +- io.swagger:swagger-models:jar:1.5.6:compile
[INFO] |  +- io.springfox:springfox-spi:jar:2.4.0:compile
[INFO] |  |  \- io.springfox:springfox-core:jar:2.4.0:compile
[INFO] |  +- io.springfox:springfox-schema:jar:2.4.0:compile
[INFO] |  +- io.springfox:springfox-swagger-common:jar:2.4.0:compile
[INFO] |  +- io.springfox:springfox-spring-web:jar:2.4.0:compile
[INFO] |  +- com.google.guava:guava:jar:18.0:compile
~~~

My environment is:

Ubuntu Linux 15.10
Oracle JDK 8u74
Maven 3.2.5

I also tried with maven wrapper: `.:mvnw` it also worked.

So, I suggest you try to cleanup your ~/.m2 local repo and/or run the maven wrapper, maybe your springfox pom is wrong.
You could also check  `mvn dependency:tree` output.

**Comment by gmarziou** [comment]
I'm closing it now and will re-open if someone else is able to reproduce it.

**Comment by imTachu** [comment]
Happened to me: Running on Windows 8, JHipster 3.0.0, jdk1.8.0_74, downloaded JHipster registry, ran it and ran the gateway generated project (no cache, h2 for development, postgresql for production, no testing tools selected, no internationalization, very plain) and got the error.

![daevidence](https://cloud.githubusercontent.com/assets/2230140/14539472/27cf88d2-0246-11e6-94b3-5a6c28514ace.png)

**Comment by imTachu** [comment]
and yes, it was fixed adding the wava dependency.

**Comment by gmarziou** [comment]
@imTachu if you want to help, please rollback and try what I proposed.

**Comment by imTachu** [comment]
@gmarziou So, I removed the guava dependency, performed a `bower update` and the issue came back. I don't know what I broke with it. I tried the `mvn dependency:tree` command you proposed with no luck :(

![image](https://cloud.githubusercontent.com/assets/2230140/14569261/3702462e-0303-11e6-99d8-848833b76241.png)

**Comment by gmarziou** [comment]
Why did you try 'bower update', your problem is in java server-side not in javascript angular.

> I tried the mvn dependency:tree command you proposed with no luck :(

What do you mean? This command is not supposed to resolve the issue, it's just a way to collect information, you were supposed to copy/paste its output.

BTW, please stop posting screenshots, just copy/paste text.

If you are skilled in these tools maybe the best would be that you share your project on github so that we can have a look.

**Comment by imTachu** [comment]
@gmarziou That wasn't an issue anymore. I tried the bower update because I have an issue while loading new functions in my AngularJS and I am basically executing everything I find in the documentation to see what helps me... When I performed that command, the problem appeared. So I tried the `mvn dependency:tree` to see if it recreated whatever happened, I just added the guava dependency anyways. It is a work project and we have a private repo, believe me I really wish I could share it with you :( 

Sorry about the screenshots, won't happen again.
